runtime/gc_structs/ConstantPoolObjectSlotIterator.cpp

```diff
@@ -61,9 +69,17 @@ GC_ConstantPoolObjectSlotIterator::nextSlot()
 
 		_cpDescription >>= J9_CP_BITS_PER_DESCRIPTION;
 		_cpDescriptionIndex -= 1;
-
-		/* Determine if the slot should be processed */
-		switch (slotType) {
+		/* If _condyOnly is TRUE, return the next constant dynamic slot in constant pool
```

Comment:
move the loop iteration variables above to the end of the loop and make them conditional based on wether we are staying in the same cpentry or not for the given state